src/build-remote/build-remote.cc

```diff
@@ -282,10 +282,21 @@ static int _main(int argc, char * * argv)
 
         uploadLock = -1;
 
-        auto drv = store->readDerivation(*drvPath);
-        drv.inputSrcs = store->parseStorePathSet(inputs);
+        BuildResult result;
 
-        auto result = sshStore->buildDerivation(*drvPath, drv);
+        BasicDerivation drv = store->readDerivation(*drvPath);
+
+        if (sshStore->isTrusting || derivationIsCA(drv.type())) {
+            drv.inputSrcs = store->parseStorePathSet(inputs);
+            result = sshStore->buildDerivation(*drvPath, drv);
+        } else {
+            try {
+                sshStore->buildPaths({{*drvPath}});
```

Comment:
Can we ensure drvPath exists on the remote ssh store?

Reply:
Yes; done.